File: CryptoScanner/test_source/weak_crypto.java
import java.security.KeyPairGenerator;
import java.security.MessageDigest;
import javax.crypto.Cipher;

public class weak_crypto {
    public static void main(String[] args) throws Exception {
        MessageDigest md1 = MessageDigest.getInstance("MD5");
        MessageDigest md2 = MessageDigest.getInstance("SHA-1");
        byte[] d = md1.digest("data".getBytes());
        Cipher c1 = Cipher.getInstance("AES/ECB/PKCS5Padding");
        Cipher c2 = Cipher.getInstance("DES/ECB/PKCS5Padding");
        KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
        kpg.initialize(1024);
        System.out.println(d.length + c1.getAlgorithm() + c2.getAlgorithm() + kpg.getAlgorithm());
    }
}

File: CryptoScanner/test_source/weak_crypto.py
import hashlib
from Crypto.Cipher import DES, ARC4

def f():
    m1 = hashlib.md5(b"data")
    m2 = hashlib.new("sha1", b"data")
    c1 = DES.new(b"12345678")
    c2 = ARC4.new(b"secretkey")
    return m1.hexdigest(), m2.hexdigest(), c1, c2

if __name__ == "__main__":
    print(f())
